### PR TITLE
Added AI

### PR DESCRIPTION
**Pull request by djshaq121**
+ AI finds the best cover around the player
+ Shoots a projectile towards target

### CODE REVIEW

File: Source/BrackeysGameJam2020/Private/EnemyAIController.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+AEnemyAIController::AEnemyAIController()
+{
+	BehaviorComp = CreateDefaultSubobject<UBehaviorTreeComponent>(TEXT("BehaviorComp"));
+	BlackboardComp = CreateDefaultSubobject<UBlackboardComponent>(TEXT("BlackboardComp"));
+
+	TargetKeyName = "Target";
+}
+
+void AEnemyAIController::OnPossess(APawn * pawn)
```
Comment (Ji-Rath) [inline Source/BrackeysGameJam2020/Private/EnemyAIController.cpp]: Just a small change to unify the code to Pascal Case.
```suggestion
void AEnemyAIController::OnPossess(APawn * Pawn)
```

File: Source/BrackeysGameJam2020/Private/EnemyAIController.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		if (AICharacter->BehaviorTree == nullptr || AICharacter->BehaviorTree->BlackboardAsset == nullptr)
+			return;
+
+		BlackboardComp->InitializeBlackboard(*(AICharacter->BehaviorTree->BlackboardAsset));
+		
+		BehaviorComp->StartTree(*AICharacter->BehaviorTree);
+	}
+}
+
+void AEnemyAIController::SetTarget(AActor* target)
```
Comment (Ji-Rath) [inline Source/BrackeysGameJam2020/Private/EnemyAIController.cpp]:
```suggestion
void AEnemyAIController::SetTarget(AActor* Target)
```

File: Source/BrackeysGameJam2020/Private/EnemyBase.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+// Called when the game starts or when spawned
+void AEnemyBase::BeginPlay()
+{
+	Super::BeginPlay();
+	
+	if (HealthComponent)
+		HealthComponent->OnHealthChange.AddDynamic(this, &AEnemyBase::OnHealthChanged);
+	
+	ACharacter* player = UGameplayStatics::GetPlayerCharacter(GetWorld(), 0);
```
Comment (Ji-Rath) [inline Source/BrackeysGameJam2020/Private/EnemyBase.cpp]:
```suggestion
	ACharacter* Player = UGameplayStatics::GetPlayerCharacter(GetWorld(), 0);
```

File: Source/BrackeysGameJam2020/Public/EnemyAIController.h [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	 UBehaviorTreeComponent* BehaviorComp;
+
+	 UBlackboardComponent* BlackboardComp;
+
+public:
+	AEnemyAIController();
+
+	virtual void OnPossess(APawn* pawn) override;
+
+	void SetTarget(AActor* target);
```
Comment (Ji-Rath) [inline Source/BrackeysGameJam2020/Public/EnemyAIController.h]:
```suggestion
	virtual void OnPossess(APawn* Pawn) override;

	void SetTarget(AActor* Target);
```

**Review by Ji-Rath — COMMENTED** [review]
Everything looks good. There are just very small inconsistencies with naming variables.

**Comment by djshaq121** [comment]
Added feedback changes, Merging to continue work on AI

**Pull request merged**